### ISSUE TITLE
Latest Version of Git Extensions Installer Always Reverts For Global user Install

### ISSUE THREAD

**Issue by mbostwick** [gitextensions/gitextensions#7120]
## Current behavior

Installer should always work, prompt for install happened today, and had a new user install it. 

## Expected behavior

Installer should work.

## Screenshots

Install latest MSI for all users. 

## Did this work in previous version of GitExtensions

3.2.0.6586.msi, I'm using 3.1, do not know which MSI

## Environment

- GitExtensions version:
- GIT version:  unrelated. 
- OS version:  Windows 10
- .NET version:  both standard user machine, and dev machine with many versions installed. 

## Diagnostics
Telemetry was not enabled during the install .

**Comment by fxlae** [comment]
The problem with rolling back also occurs with the "install for youself" option, I wasn't able to install GitExtensions-3.2.0.6586.msi

**Comment by SuperJugy** [comment]
Same Issue, clean install on Win 10.. tried all options. I saw on another bug that the telemetry option was failing the chocolatey build, maybe it is related?

**Comment by RussKie** [comment]
I need to investigate it.
I had successfully installed it on a few computers.... 🤔

**Comment by vladk** [comment]
I have same error, can't install version 3.2

**Comment by mast-eu** [comment]
Please make sure that you are installing _GitExtensions-3.2.0.**6586**.msi_.
Available here: https://github.com/gitextensions/gitextensions/releases/download/v3.2/GitExtensions-3.2.0.6586.msi

Right now, SourceForge and Chocolatey downloads still point to _GitExtensions-3.2.0.**6581**.msi_. This is NOT the final build of v3.2, as it had an issue during installation (#7106)

**Comment by fwd079** [comment]
> Please make sure that you are installing _GitExtensions-3.2.0.**6586**.msi_.
> Available here: https://github.com/gitextensions/gitextensions/releases/download/v3.2/GitExtensions-3.2.0.6586.msi
> 
> Right now, SourceForge and Chocolatey downloads still point to _GitExtensions-3.2.0.**6581**.msi_. This is NOT the final build of v3.2, as it had an issue during installation (#7106)

Yes I came here after choco failed, but this failed as well.

Getting same error in the link and MSI link in **releases** folder.

Regards.

**Comment by vladk** [comment]
> Please make sure that you are installing _GitExtensions-3.2.0.**6586**.msi_.
> Available here: https://github.com/gitextensions/gitextensions/releases/download/v3.2/GitExtensions-3.2.0.6586.msi
> 
> Right now, SourceForge and Chocolatey downloads still point to _GitExtensions-3.2.0.**6581**.msi_. This is NOT the final build of v3.2, as it had an issue during installation (#7106)

Yes, I've tried both chocolatey, and manually installing 6586 build, and it fails to install and reverts install

**Comment by RussKie** [comment]
Can someone please generate a log and publish it here?
```
msiexec /i "C:\MyPackage\GitExtensions-3.2.0.6586.msi" /L*V "C:\log\example.log"
```

**Comment by mbostwick** [comment]
Went to try this on anther machine and attach better logs if possible.  Manually clicking and running the install failed on the machine. 
`msiexec /i  GitExtensions-3.2.0.6586.msi /L*V InstallLog.log` failed
`msiexec /i  GitExtensions-3.2.0.6586.msi /L*W InstallLog.log` installed

**Comment by RussKie** [comment]
@mast-eu @maraf could it be something to do with the Plugin Manager registration?

@Johno-ACSLive do you think you could help investigate?


In one instance I got the following error:
```
WixQuietExec:  Entering WixQuietExec in C:\WINDOWS\Installer\MSI6046.tmp, version 3.11.1701.0
WixQuietExec:  "C:\Windows\SysWOW64\WindowsPowerShell\v1.0\powershell.exe" -Version 2.0 -NoProfile -NonInteractive -InputFormat None -ExecutionPolicy Bypass -Command "& 'C:\Program Files (x86)\GitExtensions\set-telemetry.ps1' 1 ; exit $($Error.Count)"
WixQuietExec:  Version v2.0.50727 of the .NET Framework is not installed and it is required to run version 2.0 of Windows PowerShell.
WixQuietExec:  Error 0xffff0000: Command line returned an error.
WixQuietExec:  Error 0xffff0000: QuietExec Failed
WixQuietExec:  Error 0xffff0000: Failed in ExecCommon method
CustomAction ConfigureTelemetry returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 1:23:57: InstallFinalize. Return value 3.
```

@mast-eu this relates to #7090. 
I now wonder whether `-Version 2.0` bit is needed at all. 🤔

@mbostwick thank you! Totally understandable. (Looks like I unintentionally used my work email for commits).
Do you have .NET 2.0 installed on your machine?

**Comment by mbostwick** [comment]
The machine I sent logs for has this :
![image](https://user-images.githubusercontent.com/547590/64681931-3300cb00-d446-11e9-8a96-4cd8197cd05b.png)
(and 2.1 .Net core installed) 

I was showing a co-worker how to use git when it failed for him, and his machine was windows 10 but beyond that it was pretty vanilla.

**Comment by RussKie** [comment]
Your verbose log contains another error:
```
Action 2:58:31: ConfigureTelemetry. 
MSI (s) (00:14) [02:58:31:727]: Executing op: CustomActionSchedule(Action=ConfigureTelemetry,ActionType=1025,Source=BinaryData,Target=WixQuietExec,CustomActionData="C:\Windows\SysWOW64\WindowsPowerShell\v1.0\powershell.exe" -Version 2.0 -NoProfile -NonInteractive -InputFormat None -ExecutionPolicy Bypass -Command "& 'C:\Program Files (x86)\GitExtensions\set-telemetry.ps1'  ; exit $($Error.Count)")
MSI (s) (00:74) [02:58:31:730]: Invoking remote custom action. DLL: C:\WINDOWS\Installer\MSI4867.tmp, Entrypoint: WixQuietExec
WixQuietExec:  Remove-Item : Cannot find path 'C:\WINDOWS\SysWOW64\&' because it does not exist.
WixQuietExec:  At C:\Program Files (x86)\GitExtensions\set-telemetry.ps1:42 char:12
WixQuietExec:  + Remove-Item <<<<  $MyINvocation.InvocationName
WixQuietExec:      + CategoryInfo          : ObjectNotFound: (C:\WINDOWS\SysWOW64\&:String) , ItemNotFoundException
WixQuietExec:      + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand
WixQuietExec:   
WixQuietExec:  Error 0x80070001: Command line returned an error.
WixQuietExec:  Error 0x80070001: QuietExec Failed
WixQuietExec:  Error 0x80070001: Failed in ExecCommon method
CustomAction ConfigureTelemetry returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 2:58:33: InstallFinalize. Return value 3.
```

🤔

**Comment by Xhanti** [comment]
Shimming in here, my complete log
[example.log](https://github.com/gitextensions/gitextensions/files/3599881/example.log)

**Comment by Johno-ACSLive** [comment]
Agree, PowerShell version 2 requires .net framework 2. Windows 10 not only has .NET 3.5 disabled by default but the payload is removed. Is there a reason to specify PowerShell version 2?

I would recommend removing this parameter in the first instance.

**Comment by RussKie** [comment]
Could you please try an msi from this build https://ci.appveyor.com/api/buildjobs/81bnavrtwlgs3vic/artifacts/combined.3.2.0.6617-unsigned.zip

**Comment by RussKie** [comment]
Could you please try the msi from this build https://ci.appveyor.com/api/buildjobs/dify5dixh6o63mba/artifacts/combined.3.2.0.6618-unsigned.zip (try #2)

**Comment by mast-eu** [comment]
> @mast-eu @maraf could it be something to do with the Plugin Manager registration?

That's a valid hypothesis, because it has been newly integrated into the v3.2 installer. However, the logs shared by @mbostwick and @Xhanti both indicate issues with PowerShell v2.0 not being available:
`WixQuietExec:  Version v2.0.50727 of the .NET Framework is not installed and it is required to run version 2.0 of Windows PowerShell.`

> @mast-eu this relates to #7090.
I now wonder whether -Version 2.0 bit is needed at all. 🤔

Probably not. Explicitly enforcing PowerShell v2.0, when v3.0 (or higher) is also available should not be necessary.
However, the above error says that even PowerShell v2.0 is not found. I do not know whether this means that there's at least PowerShell v1 available, or no PowerShell at all.
In case PowerShell v1 is available, I do not know whether the script is compatible.

**Comment by Johno-ACSLive** [comment]
> > @mast-eu @maraf could it be something to do with the Plugin Manager registration?
> 
> That's a valid hypothesis, because it has been newly integrated into the v3.2 installer. However, the logs shared by @mbostwick and @Xhanti both indicate issues with PowerShell v2.0 not being available:
> `WixQuietExec: Version v2.0.50727 of the .NET Framework is not installed and it is required to run version 2.0 of Windows PowerShell.`
> 
> > @mast-eu this relates to #7090.
> > I now wonder whether -Version 2.0 bit is needed at all. 🤔
> 
> Probably not. Explicitly enforcing PowerShell v2.0, when v3.0 (or higher) is also available should not be necessary.
> However, the above error says that even PowerShell v2.0 is not found. I do not know whether this means that there's at least PowerShell v1 available, or no PowerShell at all.
> In case PowerShell v1 is available, I do not know whether the script is compatible.

No they are failing because .NET 2.0 is not installed (which is correct for a default Win10 install). When you request PowerShell to use a particular version, each version of PowerShell targets a specific version of .NET Framework. PowerShell version 2 targets .NET Framework 2.0.

**Comment by RussKie** [comment]
I got reports that the msi from the 6618 build installs successfully. 
I'd like to get few more confirmations before I close this.

**Comment by Johno-ACSLive** [comment]
Testing this now, will provide an update shortly

**Comment by Johno-ACSLive** [comment]
OK, brand new Win10 version 1903 fully patched.

3.2.0.6586 - failed
3.2.0.6617 - failed
3.2.0.6618 - successful

**Comment by RussKie** [comment]
Awesome, thank you so much!
Martin pointed I made a repeat mistake, pushing a small fix for W7.

**Comment by Johno-ACSLive** [comment]
Not a problem!

**Comment by mbostwick** [comment]
Files attached via combined.3.2.0.6618 works for me (i.e I think we can close this issue).  Not sure if you want to close it now, or once the merge request is done.

**Comment by mstv** [comment]
I could install it successfully, too, although it failed to stop the explorer.exe.
- Git Extensions 3.2.0.6618
- Build 6861560a668a7f02bc5a04d7d31edf5579e9fbd9
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.3815.0
- DPI 96dpi (no scaling)

**Comment by mstv** [comment]
My GH credentials are not stored anymore (for https://github.com/mstv/gitextensions.git).

**Comment by fwd079** [comment]
Merging is blocked? Issue still open I think.

So should we hold fire or use the zip download link to install?

Regards.

**Comment by mast-eu** [comment]
We are iterating on #7123 to close this issue as soon as possible. Some use cases are not yet working as expected.

@mbostwick This issue will be automatically closed once we have the PR merged.

@fwd079 We are still testing. If you want to help testing the installers and provide feedback, this would be appreciated. Be aware that these are tests though, they are not yet an official release.

**Comment by fwd079** [comment]
Nope, msi in zipped archive also not working sadly.

Regards.

_
![image](https://user-images.githubusercontent.com/7912632/64855254-66468400-d617-11e9-9f7d-c38a396fad29.png)

**Comment by RussKie** [comment]
Logs?

**Comment by fwd079** [comment]
> Logs?
 
Interesting. just read this and thought okay, will rerun it to get logs as wasn't sure how to get previous logs, so ran the command given before :
`msiexec /i "C:\Users\fawad.raza\Downloads\combined.3.2.0.6618-unsigned\GitExtensions-3.2.0.6618.msi" /L*V "C:\log\example.log"`

But this time, ran **command prompt as administrator** and it all ran fine and the installation from zip was successful.

Regards.

![image](https://user-images.githubusercontent.com/7912632/64859570-7fa0fd80-d622-11e9-9147-79b07268f2a9.png)

**Comment by fwd079** [comment]
Interesting that issue was closed a minute before my reply / possible solution. But this could be something to do with permissions then?

Regards.

**Comment by RussKie** [comment]
The latest release https://github.com/gitextensions/gitextensions/releases/download/untagged-a1e07cad424acd037722/GitExtensions-3.2.1.6628.msi

**Comment by fwd079** [comment]
Thanks but link isn't working, let me go grab from Releases tab.

Regards.

**Comment by pmiossec** [comment]
@fwd079 Strange. The link is working for me. And I also see the setup in the release tab.

**Comment by pmiossec** [comment]
I understood, that's a draft release only visible by contributors.

**Comment by RussKie** [comment]
Try this https://github.com/gitextensions/gitextensions/releases/tag/v3.2.1

**Comment by fwd079** [comment]
> Try this https://github.com/gitextensions/gitextensions/releases/tag/v3.2.1

Yup this worked, thank you. 🙂

Regards

![image](https://user-images.githubusercontent.com/7912632/64871912-1f1fb980-d63e-11e9-895f-b6c43a1d0454.png)

**Comment by FEsmondeWhite** [comment]
FYI, I had the same issue (new windows 10 install).

The installer version ([GitExtensions-3.2.1.6628.msi, link](https://sourceforge.net/projects/gitextensions/files/v3.2.1/GitExtensions-3.2.1.6628.msi/download)) worked. I ran it with administrator priviledges.

FYI, https://gitextensions.github.io/ is still pointing to latest, which is GitExtensions-3.1.1.6049.msi

Thank you for making Git Extensions. It's fantastic.

RussKie, you rock.

**Comment by mstv** [comment]
> My GH credentials are not stored anymore (for mstv/gitextensions.git).

As I remember now, I have never stored the GH credentials. Hence the re-installation cannot have harmed it. The only difference is, that Git(?) does not cache the credentials in the RAM anymore, i.e. no need to open an issue. Sorry for the off-topic discussion. (I'll hide it.)

**Comment by RussKie** [comment]
FYI v3.2.1 has been published.



> Thank you for making Git Extensions. It's fantastic.

You're welcome



> RussKie, you rock.

Thank you, the credits go to every contributor making this project possible.

**Comment by fwd079** [comment]
> FYI v3.2.1 has been published.
> 
Yup, tried with Choco and that too installed successfully now.

![image](https://user-images.githubusercontent.com/7912632/65017748-613b4a80-d91f-11e9-965d-b9b59967c320.png)



Regards.

**Comment by SuperJugy** [comment]
I am having the same issue now with GitExtensions-3.2.1.6628.msi. I created a log but I don't know who to send it to. I am in my work computer so I can't publish it publicly.

**Comment by RussKie** [comment]
@SuperJugy we need more information.
What OS? Clean install or upgrade?

**Comment by SuperJugy** [comment]
Windows 10, upgrade from 3.1.1.6049

**Comment by RussKie** [comment]
Try removing C:\Users\<user>\AppData\Roaming\GitExtensions\GitExtensions
folder and try again?

**Comment by SuperJugy** [comment]
Same thing

**Comment by mast-eu** [comment]
> I created a log but I don't know who to send it to. I am in my work computer so I can't publish it publicly.

Your can send it to gitextensions@outlook.com. Only team members have access there and we won't publish the log file.

**Comment by SuperJugy** [comment]
Thanks. Sent the file. Hope it helps.

**Comment by mast-eu** [comment]
> Hope it helps.

It did. Thanks. I can reproduce your issue by setting the PowerShell language mode to "ConstrainedLanguage".

![image](https://user-images.githubusercontent.com/46861028/66412036-0b753200-e9f5-11e9-8435-05c78c0c986b.png)

Do you have a whitelisting policy enforcing this, e.g. through AppLocker or DeviceGuard?

**Comment by SuperJugy** [comment]
I am not sure where to check that, but most probably yes, they put a lot of constraints. Is there a way to make the script work in a constrained mode? or there is no way fix it and I have to see how to change the mode?

**Comment by mast-eu** [comment]
> I am not sure where to check that,

Open the Windows command line and type 
`powershell.exe $ExecutionContext.SessionState.LanguageMode` 

> Is there a way to make the script work in a constrained mode? 

I fail to see a way to make it work right now. Not saying there isn't one, I'm not a PowerShell expert.
A workaround would be to use the portable version for now. It doesn't contain the PS script, because is used only during installation.

> and I have to see how to change the mode?

If you have the possibility to change the mode (e.g. run the installer as admin), this should solve your issue.

**Comment by SuperJugy** [comment]
I checked and it is indeed in the ConstrainedLanguage mode. I can't run directly the msi as administrator (I think that is standard in all msi), but it does elevate the privileges when running, so it is running as admin. Not sure if there is a way to change the powershell mode. I'll investigate mode about this.

**Comment by mast-eu** [comment]
@SuperJugy I created a new issue for your problem, because I'm preparing a PR for it. Please have a look at #7308 and add any relevant detail that you feel might be missing.
